src/ccache.cpp

```diff
@@ -965,6 +965,12 @@ to_cache(Context& ctx,
   Util::unsetenv("DEPENDENCIES_OUTPUT");
   Util::unsetenv("SUNPRO_DEPENDENCIES");
 
+#ifndef __APPLE__
```

Comment:
A difference in compiler behavior shouldn't be solved by looking at the platform at compile time. What if an older Clang version is used on Linux or a newer Clang version is used on macOS?